src/public/media.ts

```diff
@@ -77,3 +92,268 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
+    function handleGetMediaRequest(response: string): void {
+      if (callback) {
+        const mediaResult: MediaResult = JSON.parse(response);
+        if (mediaResult.error) {
+          callback(mediaResult.error, null);
+        } else {
+          if (mediaResult.mediaChunk) {
+            // If the chunksequence number is less than equal to 0 implies EOF
+            // create file/blob when all chunks have arrived and we get 0/-1 as chunksequence number
+            if (mediaResult.mediaChunk.chunkSequence <= 0) {
+              const file = createFile(helper.assembleAttachment, helper.mediaMimeType);
+              callback(mediaResult.error, file);
+            } else {
+              // Keep pushing chunks into assemble attachment
+              const assemble: AssembleAttachment = decodeAttachment(mediaResult.mediaChunk, helper.mediaMimeType);
+              helper.assembleAttachment.push(assemble);
+            }
+          } else {
+            callback({ errorCode: ErrorCode.INTERNAL_ERROR, message: 'data receieved is null' }, null);
+          }
+        }
+      }
+    }
+
+    GlobalVars.handlers['getMedia' + actionName] = handleGetMediaRequest;
+  }
+}
+
+/**
+ * Input parameter supplied to the select Media API
+ */
+export interface MediaInputs {
+  /**
+   * Only one media type can be selected at a time
+   */
+  mediaType: MediaType;
+
+  /**
+   * max limit of media allowed to be selected in one go, current max limit is 10 set by office lens.
+   */
+  maxMediaCount: number;
+
+  /**
+   * Additional properties for customization of select media in mobile devices
+   */
+  imageProps?: ImageProps;
+}
+
+/**
+ *  All properties in ImageProps are optional and have default values in the platform
+ */
+export interface ImageProps {
+  /**
+   * Optional; Lets the developer specify the image source, more than one can be specified.
+   * Default value is both camera and gallery
+   */
+  sources?: Source[];
+
+  /**
+   * Optional; Specify in which mode the camera will be opened.
+   * Default value is Photo
+   */
+  startMode?: Mode;
+
+  /**
+   * Optional; indicate if inking on the selected Image is allowed or not
+   * Default value is true
+   */
+  ink?: boolean;
+
+  /**
+   * Optional; indicate if user is allowed to move between front and back camera
+   * Default value is true
+   */
+  cameraSwitcher?: boolean;
+
+  /**
+   * Optional; indicate if putting text stickers on the selected Image is allowed or not
+   * Default value is true
+   */
+  textSticker?: boolean;
+
+  /**
+   * Optional; indicate if image filtering mode is enabled on the selected image
+   * Default value is false
+   */
+  enableFilter?: boolean;
+}
+
+/**
+ * The modes in which camera can be launched in select Media API
+ */
+export const enum Mode {
+  Photo = 1,
+  Document = 2,
+  Whiteboard = 3,
+  BusinessCard = 4,
+}
+
+/**
+ * Specifies the image source
+ */
+export const enum Source {
```

Comment:
MediaSources ?

Reply:
This is more like imageSource because for other medias we won't have same set of sources. However as explained above not taking up renaming changes unless must have. At the time of calling it will be called up as ImageProps.sources